### ISSUE TITLE
Issue running thor-lite on centOs based machines

### ISSUE THREAD

**Issue by kraken8585** [NextronSystems/thor-lite#19]
Process terminated by Kernel using SIGKILL due to constant high CPU usage ( 100% )

Setting the CPU limit to 30 didnt make any difference as the thor executable still shows a 100% cpu usage for some reason

Info Thor Version: 10.7.3
Info Thor Build: 11e67274a424
Info Running as user: root
Info User has admin rights: yes
Info Effective argument list: **[--debug --cpulimit 30 --dbfile /var/lib/thor/thor10-lite.db]**
**Info Platform: CentOS Linux 7.9 (Core)**
Info Platform DeepEval NAME: CentOS Linux 7 (Core) KERNEL_NAME: Linux  **KERNEL_VERSION: 3.10.0-1160.76.1.el7.x86_64**  PROC: x86_64  ARCH: x86_64 
Info Language: en_US, Zone: UTC
Info CPU Count: 2
Info Memory in Megabyte: 3788

Any alternative workaround?

**Comment by Neo23x0** [comment]
We think this could happen during "ProcessCheck", which cannot be throttled during the process memory scanning. We can only make pauses between each process to limit the overall load. But if a big process memory is scanned, it will take 100% CPU for some seconds. 

You could provide us the output of

```
./thor-util diagnostics --run
```

Send it to support@nextron-systems.com

**Comment by kraken8585** [comment]
I tested in on a freshly built vanilla CentOS 7.9 box and it terminated early as well. Btw, i forgot to include that i tested this by uploading the executable via Live Response and using a bash script to execute the binary. The thor lite works well if run directly on the box.

**No issues when tested with Ubuntu boxes, whether via Live Response or directly on the box

**Comment by Neo23x0** [comment]
Are you sure that it gets terminated due to the CPU usage? 
I mean, you don't run a --quick scan, but a standard scan and Live Response restricts the script run time to IIRC 30min or even less. My guess is that the standard scan (not --quick scan) simply runs longer than this script runtime limit and gets killed. It's coincidence that the scans finish in time on the Ubuntu machines and the CentOS machines just take longer to complete.

**Comment by kraken8585** [comment]
I did a strace -p to the thor process pid and it resulted in a SIGKILL output. Researching about it online seems to point to the kernel not favoring high CPU usage and the same was reflected when I monitored the thor execution(which i launched via Live Response) directly on the box using TOP

Based on the scan debug log report, it just terminates without any error log or exit code after a minute of runtime( on average ) so it's definitely within Live Response runtime limit. 

My assumption is that Live Response + Thor Lite on CentOS box seems to be clashing with each other for some unknown reason :-)

**Comment by Neo23x0** [comment]
You could try running the scan with the `--noprocs` flag. It would skip the process memory scan and just scans the filesystem.

**Comment by kraken8585** [comment]
yes that worked! :-) hopefully we can get a fix soon. Thor-Lite rocks. Looking at the possibility of subscribing to the premium version soon

**Comment by Neo23x0** [comment]
Ping me on one of these platforms and I can send you a trial : https://linktr.ee/cyb3rops
